erts/emulator/drivers/common/inet_drv.c

```diff
@@ -8347,7 +8347,7 @@ static ErlDrvData inet_start(ErlDrvPort port, int size, int protocol)
 
 
 #ifndef MAXHOSTNAMELEN
-#define MAXHOSTNAMELEN 256
+#define MAXHOSTNAMELEN 255
```

Comment:
This is where I had in mind to make it always use 255, instead of only using 255 of `MAXHOSTNAMELEN` is not defined. This is an internal driver and the change will be limited to only here, and it is the place where the original problem occurred. Something like:

    #ifdef MAXHOSTNAMELEN
    #undef MAXHOSTNAMELEN
    #endif
    #define MAXHOSTNAMELEN 255

or maybe just introduce a new define that is used within the inet driver.

Together with a comment about why we are not using the original define.